validator/keymanager/remote-web3signer/client.go

```diff
@@ -0,0 +1,188 @@
+package remote_web3signer
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/crypto/bls"
+)
+
+const (
+	ethApiNamespace = "/api/v1/eth2/sign/"
+	maxTimeout      = 3 * time.Second
+)
+
+// HTTPClient defines interface for easier testing and mocks.
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// client a wrapper object around web3signer APIs. API docs found here https://consensys.github.io/web3signer/web3signer-eth2.html.
+type client struct {
+	BasePath   string
+	restClient HTTPClient
```

Comment:
Rather than adding additional layers of custom types, I would recommend using the standard http library patterns. http.Client is already designed to make this easy. Your test can reach into the client object and replace the `Transport` value with a custom `RoundTripper`: https://pkg.go.dev/net/http#RoundTripper

Which allows you to define a function in your tests that has access to the request value and can return any response value.

Reply:
will look int making this change